### PR TITLE
Clarify the behavior of FilePatternMatch Path and Stem

### LINKED ISSUE #36092: omitted parent directory name from FilePatternMatch.Stem.
### Describe the bug
Unexpected result was gotten in `FilePatternMatch.Stem` which from `FileSystemGlobbing.Matcher.Execute()`.

### To Reproduce
1. Using version '2.0.0' of package 'Microsoft.Extensions.FileSystemGlobbing'
2. create file `testdir/a.txt` in project dir
3. Write this code 
```c#
using System;
using System.IO;
using Microsoft.Extensions.FileSystemGlobbing;
using Microsoft.Extensions.FileSystemGlobbing.Abstractions;

namespace globtest
{
    class Program
    {
        static void Main(string[] args)
        {
            var matcher = new Matcher();
            // if change to `**/testdir/**/*`, expected results will be output.
            matcher.AddInclude("testdir/**/*");
            // base directory is current working directory
            var di = new DirectoryInfo(Directory.GetCurrentDirectory());
            var diwrapper = new DirectoryInfoWrapper(di);
            var result = matcher.Execute(diwrapper);
            foreach(var f in result.Files)
            {
                Console.WriteLine($"path = {f.Path}, stem = {f.Stem}");
            }
        }
    }
}
```
4. run with `dotnet run` in project dir

### Expected behavior
output following line;
```
path = testdir/a.txt, stem = testdir/a.txt
```

### Actual result
output following line;
```
path = testdir/a.txt, stem = a.txt
```

### Additional context
`dotnet --info` output is here;

```
.NET Core SDK (global.json を反映):
 Version:   3.0.100-preview9-014004
 Commit:    8e7ef240a5

ランタイム環境:
 OS Name:     Windows
 OS Version:  6.3.9600
 OS Platform: Windows
 RID:         win81-x64
 Base Path:   C:\Program Files\dotnet\sdk\3.0.100-preview9-014004\

Host (useful for support):
  Version: 3.0.0-preview9-19423-09
  Commit:  2be172345a

.NET Core SDKs installed:
  1.0.0-preview2-003121 [C:\Program Files\dotnet\sdk]
  1.0.0-preview2-003131 [C:\Program Files\dotnet\sdk]
  1.0.0-preview2-003133 [C:\Program Files\dotnet\sdk]
  1.0.0-preview2-003156 [C:\Program Files\dotnet\sdk]
  1.0.0-preview2-1-003177 [C:\Program Files\dotnet\sdk]
  1.0.4 [C:\Program Files\dotnet\sdk]
  2.0.0 [C:\Program Files\dotnet\sdk]
  2.0.2 [C:\Program Files\dotnet\sdk]
  2.1.2 [C:\Program Files\dotnet\sdk]
  2.1.3 [C:\Program Files\dotnet\sdk]
  2.1.4 [C:\Program Files\dotnet\sdk]
  2.1.100 [C:\Program Files\dotnet\sdk]
  2.1.104 [C:\Program Files\dotnet\sdk]
  2.1.202 [C:\Program Files\dotnet\sdk]
  2.1.300 [C:\Program Files\dotnet\sdk]
  2.1.301 [C:\Program Files\dotnet\sdk]
  2.1.400 [C:\Program Files\dotnet\sdk]
  2.1.401 [C:\Program Files\dotnet\sdk]
  2.1.402 [C:\Program Files\dotnet\sdk]
  2.1.403 [C:\Program Files\dotnet\sdk]
  2.1.500 [C:\Program Files\dotnet\sdk]
  2.1.503 [C:\Program Files\dotnet\sdk]
  2.1.505 [C:\Program Files\dotnet\sdk]
  2.1.600-preview-009472 [C:\Program Files\dotnet\sdk]
  2.1.600-preview-009497 [C:\Program Files\dotnet\sdk]
  2.1.600 [C:\Program Files\dotnet\sdk]
  2.1.602 [C:\Program Files\dotnet\sdk]
  2.1.700 [C:\Program Files\dotnet\sdk]
  2.2.100 [C:\Program Files\dotnet\sdk]
  2.2.105 [C:\Program Files\dotnet\sdk]
  2.2.200-preview-009748 [C:\Program Files\dotnet\sdk]
  2.2.200-preview-009804 [C:\Program Files\dotnet\sdk]
  2.2.200 [C:\Program Files\dotnet\sdk]
  2.2.300 [C:\Program Files\dotnet\sdk]
  3.0.100-preview9-014004 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.7 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.7 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.0.0-preview9.19424.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 1.0.0-rc2-3002702 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.0.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.0.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.0.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.1.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.1.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.3-servicing-26724-03 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.7 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.8 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.0.0-preview9-19423-09 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.0.0-preview9-19423-09 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download

```

### PR DESCRIPTION
**Pull request by jeffhandley**
Resolves #36092 by clarifying the behavior of the `Path` and `Stem` properties on `FilePatternMatch`. The current documentation and `///` comments can be interpreted differently from what is implemented. Additionally, I found the existing unit tests left this behavior ambiguous. This adds some new `[Theory]` tests that illustrate the behavior clearly and updates the `///` comments to reflect the implementation more precisely.

Once we lock in the right verbiage for the comments, I'll follow up with a PR in `dotnet/dotnet-api-docs` to fix the documentation too.

### CODE REVIEW

File: src/libraries/Microsoft.Extensions.FileSystemGlobbing/src/FilePatternMatch.cs [hunk 1]
```diff
@@ -12,29 +12,28 @@ namespace Microsoft.Extensions.FileSystemGlobbing
...
+        /// "src/Project/Interfaces/IFile.cs", then Stem = "Interfaces/IFile.cs" and Path = "src/Project/Interfaces/IFile.cs".
         /// </remarks>
         public string Stem { get; }
 
         /// <summary>
         /// Initializes new instance of <see cref="FilePatternMatch" />
         /// </summary>
         /// <param name="path">The path to the matched file</param>
-        /// <param name="stem">The stem</param>
+        /// <param name="stem">The stem path, relative to the first wildcard</param>
```
Comment (carlossanlop) [inline src/libraries/Microsoft.Extensions.FileSystemGlobbing/src/FilePatternMatch.cs:36]:
```suggestion
        /// <param name="stem">The stem path, relative to the first wildcard.</param>
```

Comment (carlossanlop) [inline src/libraries/Microsoft.Extensions.FileSystemGlobbing/src/FilePatternMatch.cs:36]:
```suggestion
        /// <param name="stem">The stem path, relative to the first wildcard.</param>
```

File: src/libraries/Microsoft.Extensions.FileSystemGlobbing/src/FilePatternMatch.cs [hunk 2]
```diff
@@ -12,29 +12,28 @@ namespace Microsoft.Extensions.FileSystemGlobbing
...
-        /// then Stem = "Interfaces/IFile.cs" and Path = "src/Project/Interfaces/IFile.cs".
+        /// If the matcher searched for "src/Project/**/*.cs" and the pattern matcher found
+        /// "src/Project/Interfaces/IFile.cs", then Stem = "Interfaces/IFile.cs" and Path = "src/Project/Interfaces/IFile.cs".
         /// </remarks>
         public string Stem { get; }
 
         /// <summary>
         /// Initializes new instance of <see cref="FilePatternMatch" />
         /// </summary>
         /// <param name="path">The path to the matched file</param>
```
Comment (carlossanlop) [inline src/libraries/Microsoft.Extensions.FileSystemGlobbing/src/FilePatternMatch.cs:35]:
```suggestion
        /// <param name="path">The path to the matched file.</param>
```

File: src/libraries/Microsoft.Extensions.FileSystemGlobbing/src/FilePatternMatch.cs [hunk 3]
```diff
@@ -12,29 +12,28 @@ namespace Microsoft.Extensions.FileSystemGlobbing
     public struct FilePatternMatch : IEquatable<FilePatternMatch>
     {
         /// <summary>
-        /// The path to the file matched
+        /// The path to the file matched, relative to the beginning of the matching search pattern
```
Comment (carlossanlop) [inline src/libraries/Microsoft.Extensions.FileSystemGlobbing/src/FilePatternMatch.cs:15]:
```suggestion
        /// The path to the file matched, relative to the beginning of the matching search pattern.
```

File: src/libraries/Microsoft.Extensions.FileSystemGlobbing/src/FilePatternMatch.cs [hunk 4]
```diff
@@ -12,29 +12,28 @@ namespace Microsoft.Extensions.FileSystemGlobbing
...
         /// </summary>
         /// <remarks>
-        /// If the matcher searched for "**/*.cs" using "src/Project" as the directory base and the pattern matcher found
+        /// If the matcher searched for "src/Project/**/*.cs" and the pattern matcher found
         /// "src/Project/Interfaces/IFile.cs", then Stem = "Interfaces/IFile.cs" and Path = "src/Project/Interfaces/IFile.cs".
```
Comment (carlossanlop) [inline src/libraries/Microsoft.Extensions.FileSystemGlobbing/src/FilePatternMatch.cs:19]:
```suggestion
        /// If the matcher searched for "src/Project/**/*.cs" and the pattern matcher found "src/Project/Interfaces/IFile.cs",
        /// then <see cref="Stem" /> = "Interfaces/IFile.cs" and <see cref="Path" /> = "src/Project/Interfaces/IFile.cs".
```

File: src/libraries/Microsoft.Extensions.FileSystemGlobbing/src/FilePatternMatch.cs [hunk 5]
```diff
@@ -12,29 +12,28 @@ namespace Microsoft.Extensions.FileSystemGlobbing
...
         /// </remarks>
         public string Path { get; }
 
         /// <summary>
-        /// The subpath to the matched file under the base directory searched
+        /// The subpath to the file matched, relative to the first wildcard in the matching search pattern
```
Comment (carlossanlop) [inline src/libraries/Microsoft.Extensions.FileSystemGlobbing/src/FilePatternMatch.cs:24]:
```suggestion
        /// The subpath to the file matched, relative to the first wildcard in the matching search pattern.
```

File: src/libraries/Microsoft.Extensions.FileSystemGlobbing/src/FilePatternMatch.cs [hunk 6]
```diff
@@ -12,29 +12,28 @@ namespace Microsoft.Extensions.FileSystemGlobbing
...
         /// </summary>
         /// <remarks>
-        /// If the matcher searched for "**/*.cs" using "src/Project" as the directory base and the pattern matcher found
-        /// "src/Project/Interfaces/IFile.cs",
-        /// then Stem = "Interfaces/IFile.cs" and Path = "src/Project/Interfaces/IFile.cs".
+        /// If the matcher searched for "src/Project/**/*.cs" and the pattern matcher found
+        /// "src/Project/Interfaces/IFile.cs", then Stem = "Interfaces/IFile.cs" and Path = "src/Project/Interfaces/IFile.cs".
```
Comment (carlossanlop) [inline src/libraries/Microsoft.Extensions.FileSystemGlobbing/src/FilePatternMatch.cs:28]:
```suggestion
        /// If the matcher searched for "src/Project/**/*.cs" and the pattern matcher found "src/Project/Interfaces/IFile.cs",
        /// then <see cref="Stem" /> = "Interfaces/IFile.cs" and <see cref="Path" /> = "src/Project/Interfaces/IFile.cs".
```

**Review by carlossanlop — COMMENTED** [review]
Thanks for the fix. I left some minor docs suggestions.

File: src/libraries/Microsoft.Extensions.FileSystemGlobbing/src/FilePatternMatch.cs (hunk 2 above)
Comment (jeffhandley) [reply src/libraries/Microsoft.Extensions.FileSystemGlobbing/src/FilePatternMatch.cs:35]: To make this more consistent:

```suggestion
        /// <param name="path">The path to the file matched, relative to the beginning of the matching search pattern.</param>
```

File: src/libraries/Microsoft.Extensions.FileSystemGlobbing/src/FilePatternMatch.cs (hunk 1 above)
Comment (jeffhandley) [reply src/libraries/Microsoft.Extensions.FileSystemGlobbing/src/FilePatternMatch.cs:36]: To make this more consistent:
```suggestion
        /// <param name="stem">The subpath to the file matched, relative to the first wildcard in the matching search pattern.</param>
```

**Review by carlossanlop — APPROVED** [review]
LGTM!

**Comment by jeffhandley** [comment]
Failure was unrelated

**Pull request merged**